presto-tests/src/test/java/com/facebook/presto/execution/resourceGroups/db/TestQueues.java

```diff
@@ -239,6 +241,34 @@ private static void cancelQuery(DistributedQueryRunner queryRunner, QueryId quer
         queryRunner.getCoordinator().getQueryManager().cancelQuery(queryId);
     }
 
+    private static void waitForCompleteQueryCount(DistributedQueryRunner queryRunner, int expectedCount)
+            throws InterruptedException
+    {
+        waitForQueryCount(queryRunner, TERMINAL_QUERY_STATES, expectedCount);
+    }
+
+    private static void waitForRunningQueryCount(DistributedQueryRunner queryRunner, int expectedCount)
+            throws InterruptedException
+    {
+        waitForQueryCount(queryRunner, ImmutableSet.of(RUNNING), expectedCount);
+    }
+
+    private static void waitForQueryCount(DistributedQueryRunner queryRunner, Set<QueryState> countingStates, int expectedCount)
+            throws InterruptedException
+    {
+        QueryManager queryManager = queryRunner.getCoordinator().getQueryManager();
+        int count;
+        do {
+            MILLISECONDS.sleep(500);
+            count = 0;
+            for (QueryInfo queryInfo : queryManager.getAllQueryInfo()) {
+                if (countingStates.contains(queryInfo.getState())) {
+                    count += 1;
+                }
+            }
+        } while (count != expectedCount);
```

Comment:
We rely on the time out ([this line](https://github.com/prestodb/presto/pull/7677/files/bda5b05ce42cd6a62ca49f5c19edafe408d89d60#diff-12e16468ce25304298fd890d92f2fea5R75)) to fail the test. Because of the race condition, I think there is no way you can use an one shot assert to check the number of running transactions.